R/Auth.R

```diff
@@ -0,0 +1,75 @@
+#' Create a connection to an instance of WebAPI
+#' 
+#' This function creates a WebAPI connection object that can be used by other
+#' functions in this package to access WebAPI. This function will test 
+#' the connection as well as trigger authentication and store the auth
+#' session token.
+#'
+#' @template baseUrl 
+#' @param authMethod The method used for authentication to WebAPI. Options are
+#' \itemize{
+#'  \item{"db"}{Database authentication using Atlas/WebAPI built in auth}
+#'  \item{"AD"}{Active Directory}
+#'  \item{"windows"}{Windows}
+#'  \item{"none"}{No authentication. WebAPI security must be disabled.}
+#' }
+#' By default the correct authentication method will be guessed.
+#' @param keyringServiceName (Optional) Keyring stores passwords securely. 
+#'                           For some authentication methods it may be convenient 
+#'                           to pass keyring service name instead of entering a password.
+#'
+#' @return A WebApi connection object
+#' @export
+connectWebApi <- function(baseUrl, 
+                          authMethod = guessAuthMethod(),
+                          atlasUsername = NULL,
+                          keyringServiceName = NULL) {
+  .checkBaseUrl(baseUrl)
+  
+  # run appropriate auth
+  if (authMethod == "db") {
```

Comment:
@anthonysena @chrisknoll may need to review this, as this should probably be the same set of auth methods supported by WebApi?